thrust/thrust/system/tbb/detail/scan.inl

```diff
@@ -250,6 +344,30 @@ inclusive_scan(tag, InputIterator first, InputIterator last, OutputIterator resu
   return result;
 }
 
+template <typename InputIterator, typename OutputIterator, typename InitialValueType, typename BinaryFunction>
+OutputIterator inclusive_scan(
+  tag, InputIterator first, InputIterator last, OutputIterator result, InitialValueType init, BinaryFunction binary_op)
+{
+  using namespace thrust::detail;
+
+  // Use the input iterator's value type per https://wg21.link/P0571
+  using ValueType = InitialValueType;
```

Comment:
@gevtushenko is this fine? According to the code review hour discussion we had yesterday, I think this should also be `accumulator_t`.

Reply:
Sorry I missed this discussion. Could either of you elaborate on it?

Reply:
We discussed that the type for the value on which to accumulate other values (the partial result) should be determined by the return type of the first application of the reduction operation. If you mentally expand a reduction, it looks like:
`binary_op(binary_op(binary_op(init, value[0]), value[1]), value[2]) ....`. Since `init` and `value[0]` can have different types, neither type may be right. It's better to use `decltype(binary_op(init, value[0]))`, or if no init exists `decltype(binary_op(value[0], value[0]))` as type holding the partial result. The CUB trait `accumulator_t` gives you this type, but it's not available in Thrust, so you may replicate it's behavior manually and add a TODO comment. We want to move the trait into libcu++, but nobody did it yet.

I pinged @gevtushenko to confirm this or state his opinion otherwise.

Reply:
fantastic